src/main/java/edu/cornell/kfs/fp/service/impl/AmazonWebServicesBillingServiceImpl.java

```diff
@@ -2,380 +2,260 @@
-                    CuFPConstants.AmazonWebServiceBillingConstants.URL_PARAMETER_MONTH + findProcessMonthNumber());
-        } catch (URISyntaxException e) {
-            throw new RuntimeException(e);
-        }
-    }
-    
-    protected List<AmazonAccountDetail> buildAmazonAcountListFromJson(String jsonResults) {
-        ObjectMapper mapper = new ObjectMapper();
-        AmazonAccountDetailContainer detailContainer = new AmazonAccountDetailContainer();
-        try {
-            detailContainer = mapper.readValue(jsonResults, AmazonAccountDetailContainer.class);
-        } catch (IOException e) {
-            throw new RuntimeException(e);
-        }
-        return detailContainer.getAccountDetail();
-    }
-
-    protected boolean processAmazonAccountDetails(List<AmazonAccountDetail> amazonAccountDetails) {
-        boolean didProcessingCompleteWithoutErrors = true;
-        AmazonBillResultsDTO resultsDTO = new AmazonBillResultsDTO();
-        for (AmazonAccountDetail amazonAccountDetail : amazonAccountDetails) {
-            if (StringUtils.equalsIgnoreCase(CuFPConstants.AmazonWebServiceBillingConstants.INTERNAL_KFS_ACCOUNT_DESCRIPTION, amazonAccountDetail.getKfsAccount())) {
-                LOG.info("processAmazonAccountDetails() Found an internal kfs account, ignoring this transaction for AWS account " 
-                    + amazonAccountDetail.getAwsAccount());
-                resultsDTO.internalCount++;
-            } else {
-                didProcessingCompleteWithoutErrors = processExternalBill(amazonAccountDetail, resultsDTO) && didProcessingCompleteWithoutErrors;
-            }
-        }
-        logSummaryReport(resultsDTO);
-        return didProcessingCompleteWithoutErrors;
-    }
-
-    private boolean processExternalBill(AmazonAccountDetail amazonAccountDetail, AmazonBillResultsDTO resultsDTO) {
-        boolean valid = true;
-        try {
-            AmazonBillingDistributionOfIncomeTransactionDTO transactionDTO = buildTransactionDTO(amazonAccountDetail);
-            if (transactionDTO.isTransactionInputError()) {
-                valid = false;
-                LOG.error("processAmazonAccountDetails() there errors that prevent a creation of a DI " + amazonAccountDetail.toString());
-                resultsDTO.erroredCount++;
-            } else if (shouldBuildDistributionOfIncomeDocument(amazonAccountDetail, transactionDTO.getAmount())) {
-                buidAndRouteDistributionOfIncomeDocument(amazonAccountDetail, transactionDTO, resultsDTO);
-            } else {
-                resultsDTO.existingDICount++;
-            }
-        } catch (Throwable e) {
-            valid = false;
-            LOG.error("processAmazonAccountDetails() there was a problem building a DI for " + amazonAccountDetail.toString(), e);
-            GlobalVariables.getMessageMap().getErrorMessages().clear();
-            resultsDTO.erroredCount++;
+        logResults(resultsDTO);
+        resetProperties();
```

Comment:
Should this resetProperties() call be moved to before the ID generation (or have some additional "finally" handling be introduced to force the call), in case an unhandled exception causes the call to be skipped?